pkg/internal/source/kind.go

```diff
@@ -53,7 +53,7 @@ func (ks *Kind) Start(ctx context.Context, handler handler.EventHandler, queue w
 
 		// Tries to get an informer until it returns true,
 		// an error or the specified context is cancelled or expired.
-		if err := wait.PollImmediateUntilWithContext(ctx, 10*time.Second, func(ctx context.Context) (bool, error) {
+		if err := wait.PollUntilContextCancel(ctx, 10*time.Second, true, func(ctx context.Context) (bool, error) {
```

Comment:
Why was this changed?

Reply:
`PollImmediateUntilWithContext` is deprecated

Reply:
AFAICS `PollWithContextCancel` is not defined. I wonder if it's a typo because the commit adds `PollUntilContextCancel` https://github.com/kubernetes/apimachinery/commit/831cf05a7c4f839939a6c5e1e8bee238ddc6baac

Reply:
It seems a typo issue.

Reply:
@mythi Looks like it was incorrect, but they didn't mean `PollUntilContextCancel`, they meant `PollUntilContextTimeout` (https://github.com/kubernetes/kubernetes/commit/1cce3b3e2f5981a6cdc2d8ead921538533399575)

Reply:
@sbueringer thanks for the update!